fe/fe-core/src/main/java/com/starrocks/load/routineload/KafkaTaskInfo.java

```diff
@@ -169,8 +170,13 @@ public TRoutineLoadTask createRoutineLoadTask() throws UserException {
 
     @Override
     protected String getTaskDataSourceProperties() {
-        Gson gson = new Gson();
-        return gson.toJson(partitionIdToOffset);
+        StringJoiner joiner = new StringJoiner(",", "{", "}");
```

Comment:
1. It's better to use a sorted ConcurrentMap like ConcurrentSkipListMap. Every time gets routine load   
properties with a sort are consuming.
2. It's better to describe the effect of the refactor in the commit message